cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -123,6 +140,18 @@ TYPED_TEST(TestNumericSumKernel, SimpleSum) {
   ValidateSum<TypeParam>(&this->ctx_, "[0, 1, 2, 3, 4, 5]",
                          Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
 
+  std::vector<std::string> json{"[0, 1, 2, 3, 4, 5]"};
+  ValidateSum<TypeParam>(&this->ctx_, json,
+                         Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
+
+  json = {"[0, 1, 2]", "[3, 4, 5]"};
+  ValidateSum<TypeParam>(&this->ctx_, json,
+                         Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
+
+  json = {"[0, 1, 2]", "[]", "[3, 4, 5]"};
+  ValidateSum<TypeParam>(&this->ctx_, json,
+                         Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
+
```

Comment:
How about using literal instead of defining a variable?
(BTW, `json` isn't a good variable name because it's an array of JSON. `jsons` or something will be better.)

```diff
diff --git a/cpp/src/arrow/compute/kernels/aggregate_test.cc b/cpp/src/arrow/compute/kernels/aggregate_test.cc
index 3e49e6bc9..7b606d75a 100644
--- a/cpp/src/arrow/compute/kernels/aggregate_test.cc
+++ b/cpp/src/arrow/compute/kernels/aggregate_test.cc
@@ -140,16 +140,13 @@ TYPED_TEST(TestNumericSumKernel, SimpleSum) {
   ValidateSum<TypeParam>(&this->ctx_, "[0, 1, 2, 3, 4, 5]",
                          Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
 
-  std::vector<std::string> json{"[0, 1, 2, 3, 4, 5]"};
-  ValidateSum<TypeParam>(&this->ctx_, json,
+  ValidateSum<TypeParam>(&this->ctx_, {"[0, 1, 2, 3, 4, 5]"},
                          Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
 
-  json = {"[0, 1, 2]", "[3, 4, 5]"};
-  ValidateSum<TypeParam>(&this->ctx_, json,
+  ValidateSum<TypeParam>(&this->ctx_, {"[0, 1, 2]", "[3, 4, 5]"},
                          Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
 
-  json = {"[0, 1, 2]", "[]", "[3, 4, 5]"};
-  ValidateSum<TypeParam>(&this->ctx_, json,
+  ValidateSum<TypeParam>(&this->ctx_, {"[0, 1, 2]", "[]", "[3, 4, 5]"},
                          Datum(std::make_shared<ScalarType>(static_cast<T>(5 * 6 / 2))));
 
   const T expected_result = static_cast<T>(14);
```